### PR TITLE
[feat-48818]: hook for making adjustments of tx deploy value

### PR DESCRIPTION
**Pull request by maryjanyes**
closes https://github.com/web3-nomad/accelerator-defi-EIP-ui/issues/89

### CODE REVIEW

File: src/hooks/eip4626/mutations/useDeployVault.ts [hunk 1]
```diff
@@ -43,12 +51,18 @@ export function useDeployVault() {
...
       );
 
       //@TODO clear query cache for the list of vaults
       return deployResult;
     },
   });
+
+  useEffect(() => {
+    setIsTxError(mut.isError);
+  }, [mut.isError]);
```
Comment (dchernokur) [inline src/hooks/eip4626/mutations/useDeployVault.ts:65]: why have separate field for the error, can we use the error from mutation directly?

File: src/hooks/useDeployValueSafeTx.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export function useDeployValueSafeTx(
+  currencyName: string,
+  fiatCurrencyName: string,
+  initialFiatAmount: number,
+  txSubmitError?: boolean,
+) {
+  const [tokenRate, setTokenRate] = useState<number>();
+  const [currentDeployValue, setCurrentDeployValue] = useState<number>(0);
+  const [currentDeployValueParsed, setCurrentDeployValueParsed] =
+    useState<string>("0");
```
Comment (dchernokur) [inline src/hooks/useDeployValueSafeTx.ts:13]: no need for explicit typings, if we have default values

File: src/hooks/useDeployValueSafeTx.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  useEffect(() => {
+    if (txSubmitError) {
+      setCurrentDeployValue((prev) => (prev ?? 0) + 1);
+    }
+  }, [txSubmitError]);
+
+  useEffect(() => {
+    getFiatCurrencyRate(currencyName, fiatCurrencyName).then((rateResponse) => {
+      rateResponse.json().then((rateData) => {
```
Comment (dchernokur) [inline src/hooks/useDeployValueSafeTx.ts:29]: why not async/await?

**Review by dchernokur — CHANGES_REQUESTED** [review]

File: src/hooks/eip4626/mutations/useDeployVault.ts [hunk 4]
```diff
@@ -3,15 +3,23 @@ import { useMutation } from "@tanstack/react-query";
...
 export function useDeployVault() {
   const { accountEvm, walletInterface } = useWalletInterface();
 
-  return useMutation({
+  const [isTxError, setIsTxError] = useState<boolean>();
+
+  const { currentDeployValue, currentDeployValueParsed } = useDeployValueSafeTx(
+    "hedera-hashgraph",
+    "usd",
+    0.9,
```
[the remark this answers is not in the document]
Comment (maryjanyes) [reply src/hooks/eip4626/mutations/useDeployVault.ts:18]: > what is this 0.9 value and where it comes from? do we need a const with telling name for it maybe?

Yeah, good proposal. It's about initial value in fiat.

**Review by dchernokur — APPROVED** [review]

**Pull request merged**